docs/api-reference/types/event.md

```diff
@@ -0,0 +1,208 @@
+    {
+        MsgId = Guid.NewGuid();
+        CorrelationId = MsgId;
+        CausationId = MsgId;
+    }
+    
+    protected Event(Guid correlationId, Guid causationId)
+    {
+        MsgId = Guid.NewGuid();
+        CorrelationId = correlationId;
+        CausationId = causationId;
+    }
+}
+```
+
+## Key Features
+
+- **Message Identity**: Provides a unique `MsgId` for each event
+- **Correlation Tracking**: Implements `ICorrelatedMessage` for tracking related messages
+- **Immutability**: Ensures events are immutable after creation
+- **Type Safety**: Provides a type-safe base for all event implementations
+
+## Usage
+
+### Defining an Event
+
+To create a new event type, inherit from the `Event` base class:
+
+```csharp
+public class AccountCreated : Event
+{
+    public readonly Guid AccountId;
+    public readonly string AccountNumber;
+    public readonly string CustomerName;
+    
+    public AccountCreated(Guid accountId, string accountNumber, string customerName)
+        : base()
+    {
+        AccountId = accountId;
+        AccountNumber = accountNumber;
+        CustomerName = customerName;
+    }
+    
+    // Constructor for correlated events
+    public AccountCreated(Guid accountId, string accountNumber, string customerName, 
+                         Guid correlationId, Guid causationId)
+        : base(correlationId, causationId)
+    {
+        AccountId = accountId;
+        AccountNumber = accountNumber;
+        CustomerName = customerName;
+    }
+}
+```
+
+### Using MessageBuilder with Events
+
+It's recommended to use the `MessageBuilder` factory to create events with proper correlation:
+
+```csharp
+// Create an event from a command
+ICorrelatedMessage command = // ... existing command
+var createdEvent = MessageBuilder.From(command, () => new AccountCreated(
+    Guid.NewGuid(), 
+    "ACC-123", 
+    "John Doe"
+));
+```
+
+### Handling Events
+
+Events are typically handled by event handlers:
+
+```csharp
+public class AccountCreatedHandler : IEventHandler<AccountCreated>
+{
+    private readonly IReadModelRepository<AccountSummary> _repository;
+    
+    public AccountCreatedHandler(IReadModelRepository<AccountSummary> repository)
+    {
+        _repository = repository;
+    }
+    
+    public void Handle(AccountCreated @event)
+    {
+        var accountSummary = new AccountSummary(@event.AccountId);
+        accountSummary.Update(@event.AccountNumber, @event.CustomerName, 0);
+        _repository.Save(accountSummary);
+    }
+}
+```
+
+## Integration with Aggregates
+
+Events are produced by aggregates in response to commands:
+
+```csharp
+public class Account : AggregateRoot
+{
+    private string _accountNumber;
+    private string _customerName;
+    private decimal _balance;
+    
+    public Account(Guid id, ICorrelatedMessage source) : base(id)
+    {
+        Apply(MessageBuilder.From(source, () => new AccountCreated(id, "ACC-" + id.ToString().Substring(0, 8), "New Customer")));
```

Comment:
This should be `Raise(new AccountCreated(...))`.

`Apply` methods are used for rehydration, not for creating new events.